src/DefaultBuilder/src/WebApplicationOptions.cs

```diff
@@ -64,5 +64,38 @@ internal void ApplyHostConfiguration(IConfigurationBuilder builder)
                 builder.AddInMemoryCollection(config);
             }
         }
+
+        internal void ApplyApplicationName(IWebHostBuilder webHostBuilder)
+        {
+            string? applicationName = null;
+
+            if (ApplicationName is not null)
+            {
+                applicationName = ApplicationName;
+            }
+
+            // We need to "parse" the args here since
+            // we need to set the application name via UseSetting
+            if (Args is not null)
```

Comment:
Why does `Args["applicationName"]` override `ApplicationName`?

Reply:
I thought we had args win over the property. I'll swap it back.

Reply:
👍 We already try applying explicit options after args [here](https://github.com/dotnet/aspnetcore/blob/1fc592c683c97411b7e6ff6116370d6106f0c330/src/DefaultBuilder/src/WebApplicationBuilder.cs#L77-L78).

Reply:
Fixed.